charts/DataTableTransform.ts

```diff
@@ -108,15 +110,65 @@ function getHeaderUnit(unit: string) {
 export class DataTableTransform extends ChartTransform {
     chart: ChartConfig
 
-    @computed get isValidConfig(): boolean {
-        return true
-    }
-
     constructor(chart: ChartConfig) {
         super(chart)
         this.chart = chart
     }
 
+    @computed private get loadedWithData(): boolean {
+        return this.dimensions.length > 0
+    }
+
+    private readonly AUTO_SELECTION_THRESHOLD_PERCENTAGE: number = 0.5
+
+    /**
+     * If the user or the editor hasn't specified a start, auto-select a start year
+     *  where AUTO_SELECTION_THRESHOLD_PERCENTAGE of the entities have values.
+     */
+    @computed get autoSelectedStartYear(): number | undefined {
+        let autoSelectedStartYear: number | undefined = undefined
+
+        if (
+            this.chart.userHasSetTimeline ||
+            this.initialTimelineStartYearSpecified ||
+            !this.loadedWithData
+        )
+            return undefined
+
+        const numEntitiesInTable = this.entities.length
+
```

Comment:
You could do an ~O(N) algo along the lines of this: 
`groups = groupBy(this.chart.table.unfilteredRows, ["year", dataTableColumn])
sorted = sortBy(groups, group => group[0].year)
entityCount = this.chart.table.entities // there is some computed here with that
autoSelectedStartYear = sorted.find(group => group.length /entityCount > AUTO_SELECTION_THRESHOLD_PERCENTAGE `

Reply:
There's probably even a cleaner way to do it, and perhaps a general method we could add to the table class for these sorts of pivots.

Reply:
Hmm, wouldn't that break on the Data Explorer (or at least require a special case) because the underlying Table includes data not used by the DataTable?

Reply:
Yes as I wrote it above. Looking at yours again I'm not sure if algo speed is too important, I was thinking there were a lot of dimensions but usually just a few.

Could be fine as is, but I think this would be a good candidate for a method on the Table class, which can then be easily unit tested. Maybe something like:

```
testCase = `year,gdp,country
2009,123,brazil
2010,456,brazil
2009,122,usa
2011,333,france`

const table = new Table(testCase)
const counts = table.getCountsByYear()
console.log(counts)
/*
{2009: {gdp: 2},
2010: {gdp: 1},
2011: {gdp: 1}
*/
```